packages/next/build/webpack/plugins/middleware-plugin.ts

```diff
@@ -141,7 +142,10 @@ export default class MiddlewarePlugin {
           envPerRoute.clear()
 
           for (const [name, info] of compilation.entries) {
-            if (name.match(MIDDLEWARE_ROUTE)) {
+            if (
+              info.options.runtime === MIDDLEWARE_SSR_RUNTIME_WEBPACK ||
+              name.match(MIDDLEWARE_ROUTE)
```

Comment:
```suggestion
              info.options.runtime === MIDDLEWARE_RUNTIME_WEBPACK
```

for consistency I would check the runtime in both cases